sdk/python/kfp/cli/cli_test.py

```diff
@@ -166,6 +166,82 @@ def test_deprecation_warning(self):
                       res.stdout.decode('utf-8'))
 
 
+class TestKfpDslCompile(unittest.TestCase):
+
+    def invoke(self, args):
+        starting_args = ['dsl', 'compile']
+        args = starting_args + args
+        runner = testing.CliRunner()
+        return runner.invoke(
+            cli=cli.cli, args=args, catch_exceptions=False, obj={})
+
+    def test_compile_with_caching_flag_enabled(self):
+        with tempfile.NamedTemporaryFile(suffix='.py') as temp_pipeline:
+            # Write the pipeline function to a temporary file
+            temp_pipeline.write(b"""
+from kfp import dsl
+
+@dsl.component
+def my_component():
+    pass
+
+@dsl.pipeline(name="tiny-pipeline")
+def my_pipeline():
+    my_component_task = my_component()
+""")
+            temp_pipeline.flush()  # Ensure the data is written to disk
+
+            # Invoke the CLI command with the temporary file
+            result = self.invoke(
+                ['--py', temp_pipeline.name, '--output', 'test_output.yaml'])
+            print(result.output)  # Print the command output
+            self.assertEqual(result.exit_code, 0)
```

Comment:
It would probably be good to have some additional/more vigorous validation instead of just a return code check here.  Perhaps some checks on the contents of `test_output.yaml`?

Reply:
Granted, this is for testing the CLI, not the compiler itself, so it likely could be something lightweight, even if it's just 'Check for existence of test_output.yaml and ensure it's not empty' to verify the command actually did something

Reply:
Done, added some checks on the contents of the generated yamls.